src/node_url.cc

```diff
@@ -162,23 +170,17 @@ namespace url {
               } else {
                 value = value * 10 + number;
               }
-              if (value > 255)
-                goto end;
               pointer++;
               ch = pointer < end ? pointer[0] : kEOL;
+              if (value > 255)
+                goto end;
```

Comment:
Oh, I was just reading the spec from the top to the bottom. The order comes from it, but yeah it's better to not touch for the performance. I will update it :)